app/javascript/react/src/containers/Locations.js

```diff
@@ -0,0 +1,83 @@
+import React, { Component } from 'react';
+import LocationForm from './LocationForm';
+import { Link } from 'react-router-dom';
+
+class Locations extends Component {
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      locations: [],
+      formShow: false
+    }
+
+    this.handleFormShow = this.handleFormShow.bind(this)
+    this.createLocation = this.createLocation.bind(this);
+  }
+
+  componentDidMount() {
+    fetch('/api/v1/locations')
+    .then(response => {
+      return response.json()
+    })
+    .then(body => {
+      this.setState({ locations: body })
+    })
+  }
+
+  createLocation(payload) {
+    console.log(`payload: ${payload}`)
+
+    fetch('/api/v1/locations', {
+      method: 'POST',
+      credentials: "same-origin",
+      body: JSON.stringify(payload)
+    }).then(response => {
+      let body = response.json()
```

Comment:
Minor point, but you can just `return response.json()`, or even do a single line (yay ES6!)

`.then(response => response.json())`

which does an implicit return